cpp/src/arrow/compute/kernels/scalar_cast_nested.cc

```diff
@@ -84,6 +84,13 @@ struct CastList {
                                                       options, ctx->exec_context()));
 
         out_scalar->is_valid = true;
+
+        if (is_fixed_size_list_type<DestType>::value) {
```

Comment:
Normally, we do not use SFINAE aliases for declarative runtime checks
```c++
if (out_scalar->type->id() == Type::FIXED_SIZE_LIST) { ... }
```

Nevertheless, the code is not reaching the Scalar block (from pyarrow) bc Scalar cast does not works (see [JIRA comment](https://issues.apache.org/jira/browse/ARROW-15455?focusedCommentId=17500385&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17500385)).